### ISSUE TITLE
DisplayServer.WindowSetVsyncMode() causes crash

### ISSUE THREAD

**Issue by stephen-soul** [godotengine/godot#65391]
### Godot version

v4.0.alpha.mono.custom_build [e7a0a97c0]

### System information

EndeavourOS Linux

### Issue description

I've observed odd behaviour related to calling `DisplayServer.WindowSetVsyncMode(DisplayServer.VSyncMode.Enabled);`. This seemingly works as intended if not done through a button. I'll attach a project using a toggle button but a similar behaviour is found with a MenuButton. Calling this function through the `_Ready` function seems to launch without crashing, whereas setting this through causes the below image, or it causes the screen to freeze, or the game to just crash.

The editor returns this error
```
E 0:00:01:0859   void Godot.DelegateUtils.InvokeWithVariantArgs(IntPtr , Godot.NativeInterop.godot_variant** , UInt32 , Godot.NativeInterop.godot_variant* ): Condition "swapchainImageCount != sp_image_count" is true. Returning: ERR_BUG
  <C++ Source>   drivers/vulkan/vulkan_context.cpp:1756 @ _update_swap_chain()
  <Stack Trace>  Godot.NativeInterop.NativeFuncs.generated.cs:324 @ void Godot.NativeInterop.NativeFuncs.godotsharp_method_bind_ptrcall(IntPtr , IntPtr , System.Void** , System.Void* )()
                 NativeCalls.cs:548 @ void Godot.NativeCalls.godot_icall_2_62(IntPtr , IntPtr , Int32 , Int32 )()
                 DisplayServer.cs:1716 @ void Godot.DisplayServer.WindowSetVsyncMode(Godot.DisplayServer+VSyncMode , Int32 )()
                 CheckButtonVsync.cs:14 @ void CheckButtonVsync.OnToggled(Boolean )()
                 :0 @ System.Object System.RuntimeMethodHandle.InvokeMethod(System.Object , System.Span`1[System.Object]& , System.Signature , Boolean , Boolean )()
                 :0 @ System.Object System.Reflection.RuntimeMethodInfo.Invoke(System.Object , System.Reflection.BindingFlags , System.Reflection.Binder , System.Object[] , System.Globalization.CultureInfo )()
                 :0 @ System.Object System.Delegate.DynamicInvokeImpl(System.Object[] )()
                 :0 @ System.Object System.Delegate.DynamicInvoke(System.Object[] )()
                 DelegateUtils.cs:57 @ void Godot.DelegateUtils.InvokeWithVariantArgs(IntPtr , Godot.NativeInterop.godot_variant** , UInt32 , Godot.NativeInterop.godot_variant* )()

```

![image](https://user-images.githubusercontent.com/29432305/188531066-05c0899a-64b3-4ca3-beb2-18868fccdee5.png)

### Steps to reproduce

- Open the attached project
- Run the project
- Toggle the Toggle Button.
- Observe 1 of 3 noted problematic scenarios

### Minimal reproduction project

[Vsync_crash.zip](https://github.com/godotengine/godot/files/9492750/Vsync_crash.zip)

**Comment by Calinou** [comment]
Can you reproduce this in a GDScript project? This is *probably* not C#-related, but just to make sure.

**Comment by stephen-soul** [comment]
I've tested in v4.0.alpha.custom_build [b6d102c7c] (another build I had that was non-mono. I can try another build if required) and observed the same behaviour with GDScript, as suspected.
[vsync_gdscript.zip](https://github.com/godotengine/godot/files/9493195/vsync_gdscript.zip)

**Comment by BluesM18A1** [comment]
I was about to open a new issue about this but my experience was slightly different. In my case it only froze with that garbled screen if I set it to mailbox mode, or if the project settings started in mailbox mode and changed to something else.

I am also using EndeavourOS Linux, running Godot 4.0 RC3.
and my GPU is an AMD Radeon 5700XT if that information helps narrow down the issue.

> Can you reproduce this in a GDScript project? This is _probably_ not C#-related, but just to make sure.

I can also confirm this is not a C# issue, I have tried both C# and GDscript.
